utils/fasm2frames.py

```diff
@@ -54,10 +54,56 @@ def dump_frm(f, frames):
             '0x%08X ' % addr + ','.join(['0x%08X' % w for w in words]) + '\n')
 
 
-def run(db_root, filename_in, f_out, sparse=False, roi=None, debug=False):
+def find_pudc_b(db):
+    """ Find PUDC_B pin func in grid, and return the tile and site prefix.
+
+    The PUDC_B pin is a special 7-series pin that controls unused pin pullup.
+
+    If the PUDC_B is unused, it is configured as an input with a PULLUP.
```

Comment:
This is my understanding;

 * The PUDC_B pin controls unused pin pullup state by the value being samples during start up sequence. 
 * If you don't have an external pullup or pulldown on this pin you need to configure it with an internal PULLUP / PULLDOWN so that it isn't randomly floating when sampled (and hence you get a repeatable unused pin pullup state).

Reply:
Not exactly.  The documentation specifically says to **physically** pull up or pull down the signal.  It is not exactly clear why Vivado emits bitstreams with a PULLUP, but I'd guess power savings.